src/Orleans.CodeGenerator/CopierGenerator.cs

```diff
@@ -361,7 +371,7 @@ static CopierFieldDescription GetCopierDescription(IMemberDescription member, in
                                 })))));
                 }
             }
-            else
+            else if (!onlyDeepFields)
```

Comment:
Does this optimization yield a perf gain

Reply:
Yes, for larger structs it can be quite significant. Previously a empty structure was created and then filled in field-by-field, but now only fields that actually need deep copies are copied and even those fields are updated in-place.